### PR TITLE
Granule search api

### PR DESCRIPTION
**Pull request by williamh890**

### CODE REVIEW

File: lambdas/find_new_granules/src/lambda_function.py [hunk 1]
```diff
@@ -14,17 +14,24 @@ def lambda_handler(event, context):
     """
     setup_env()
 
-    search_results = find_new.granules()
-    new_granule_events = results.package(search_results)
+    new_granule_events = find_new.granules()
```
Comment (Askaholic) [inline lambdas/find_new_granules/src/lambda_function.py]: Function name could be improved. What is `granules()` supposed to do. Maybe it returns some granules? But then it's assigned to `new_granule_events` so does that mean it returns an event? or a list of events? And what sort of granules are they? The module name helps with this, but what if the module name changes?

File: lambdas/find_new_granules/src/lambda_function.py [hunk 2]
```diff
@@ -14,17 +14,24 @@ def lambda_handler(event, context):
...
 
     if not any_new_granules(new_granule_events):
         print('No new granules. Done.')
         return
 
-    events_json = results.format_into_json(new_granule_events)
+    events_json = format_into_json(new_granule_events)
```
Comment (Askaholic) [inline lambdas/find_new_granules/src/lambda_function.py]: maybe `format_as_json()`?

File: modules/granule_search/setup.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    if '-i https://pypi.org/simple' in requirements[0]:
+        requirements = requirements[1:]
+
+setup(
+    name='Granule Search',
+    version='0.1dev',
+    packages=find_packages('src'),
+    package_dir={'': 'src'},
+    license="Wrappers around granule search api's",
```
Comment (Askaholic) [inline modules/granule_search/setup.py]: That's not a license

File: modules/granule_search/src/granule_search/cmr/__init__.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+"""CMR module for searching granules """
```
Comment (Askaholic) [inline modules/granule_search/src/granule_search/cmr/__init__.py]: Add space after `"""`

File: modules/granule_search/src/granule_search/cmr/results/package.py [hunk 5]
```diff
@@ -77,17 +75,11 @@ def is_granule_in(title):
 
 def make_granule_from(result):
     name = result['producer_granule_id']
+    print(f'found granule: {name}')
```
Comment (Askaholic) [inline modules/granule_search/src/granule_search/cmr/results/package.py:78]: Left over debug printout?

File: modules/granule_search/src/granule_search/search_api.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
+import abc
+
+
+class GranuleSearchAPI(abc.ABC):
+    """A generic interface for granule search api's"""
+    @property
+    @abc.abstractmethod
+    def api_url(self):
+        """Base url for the api"""
```
Comment (Askaholic) [inline modules/granule_search/src/granule_search/search_api.py]: Add space before `"""`

File: modules/granule_search/tests/test_cmr_searchs.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import datetime as dt
```
Comment (Askaholic) [inline modules/granule_search/tests/test_cmr_searchs.py]: 'searches' has an `e` in it (in reference to the file name)

**Review by Askaholic — COMMENTED** [review]
Couple of formatting suggestions

File: modules/granule_search/src/granule_search/cmr/results/package.py (hunk 5 above)
Comment (williamh890) [reply modules/granule_search/src/granule_search/cmr/results/package.py:78]: Put that there for logging in the lambda

**Pull request merged**